### PR TITLE
Fix misc problems in test code and webpack 4

### PR DESCRIPTION
**Pull request by leemars**
I fixed two problems when I run `npm run test`.

* assert.throws problem of parseQuery test
* eslint problems

### CODE REVIEW

**Comment by leemars** [comment]
And I fixed two problems in webpack 4.

* this.options was replaced by this.query in webpack 4
* this.options.context was replaced by this.rootContext in webpack 4

Refs:
https://webpack.js.org/api/loaders/#this-options
https://webpack.js.org/api/loaders/#this-rootcontext

**Comment by leemars** [comment]
Any progress?

File: lib/stringifyRequest.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function isRelativePath(str) {
 
 function stringifyRequest(loaderContext, request) {
 	const splitted = request.split("!");
-	const context = loaderContext.context || (loaderContext.options && loaderContext.options.context);
+	const context = loaderContext.context || loaderContext.rootContext || (loaderContext.options && loaderContext.options.context);
```
Comment (evilebottnawi) [inline lib/stringifyRequest.js]: `rootContext` and `context` are difference, and it is invalid fix, `rootContext` is context for request, `context` is `webpack` context

File: lib/interpolateName.js [hunk 2]
```diff
@@ -90,8 +91,9 @@ function interpolateName(loaderContext, name, options) {
 			);
 		});
 	}
-	if(typeof loaderContext.options === "object" && typeof loaderContext.options.customInterpolateName === "function") {
-		url = loaderContext.options.customInterpolateName.call(loaderContext, url, name, options);
+	const loaderContextOptions = getOptions(loaderContext);
```
Comment (evilebottnawi) [inline lib/interpolateName.js]: Already fixed

**Comment by evilebottnawi** [comment]
Thanks for the PR, some problem already fixed, some invalid

File: lib/stringifyRequest.js (hunk 1 above)
Comment (leemars) [reply lib/stringifyRequest.js]: But according to [this](https://webpack.js.org/api/loaders/#this-rootcontext), `this.options.context` has been changed to `this.rootContext`. I thought `this` is just the `loaderContext`. Could you give me some more explanation? Thanks!

Comment (evilebottnawi) [reply lib/stringifyRequest.js]: hm, looks my mistake, can you create new a PR?

File: lib/interpolateName.js (hunk 2 above)
Comment (leemars) [reply lib/interpolateName.js]: Any updates?

**Pull request closed without merging**